src/hotspot/share/adlc/adlArena.cpp

```diff
@@ -147,8 +149,9 @@ void *AdlArena::Arealloc( void *old_ptr, size_t old_size, size_t new_size ) {
   }
 
   // See if we can resize in-place
-  if( (c_old+old_size == _hwm) &&       // Adjusting recent thing
-      (c_old+new_size <= _max) ) {      // Still fits where it sits
+  if( (c_old+old_size == _hwm) &&            // Adjusting recent thing
+      ((size_t)(_max-c_old) >= new_size) ) { // Still fits where it sits, safe from overflow
+
```

Comment:
This code appears to be a copy of Arena::Arealloc, so we should probably fix both at the same time.

Reply:
Good catch! Fixed in cf0b434.